isce2_topsapp/localize_slc.py

```diff
@@ -232,3 +223,27 @@ def download_one(resp):
         "reference_properties": reference_props,
         "secondary_properties": secondary_props,
     }
+
+
+def _get_frame_by_id(frame_id: int) -> gpd.GeoSeries:
+    frames = gpd.read_file(Path(__file__).parent / 'data' / 's1_frames_latitude_aligned.geojson.zip')
+    return frames[frames.frame_id == frame_id].reset_index(drop=True).iloc[0]
+
+
+def get_slcs_for_date_and_frame(date: datetime.date, frame_id: int) -> list[str]:
+    frame = _get_frame_by_id(frame_id)
+    date_as_datetime = datetime.datetime(year=date.year, month=date.month, day=date.day)
+    results = asf.search(
+        dataset=asf.constants.DATASET.SENTINEL1,
+        processingLevel=asf.constants.PRODUCT_TYPE.SLC,
+        beamMode=asf.constants.BEAMMODE.IW,
+        polarization=[asf.constants.POLARIZATION.VV, asf.constants.POLARIZATION.VV_VH],
+        flightDirection=frame.orbit_direction,
+        relativeOrbit=list({frame.relative_orbit_number_min, frame.relative_orbit_number_max}),
+        intersectsWith=frame.geometry.wkt,
+        start=date_as_datetime - datetime.timedelta(minutes=5),
+        end=date_as_datetime + datetime.timedelta(days=1, minutes=5),
```

Comment:
There may be an edge case here in which only scenes from just before or after the provided date are found, but none on the provided date itself.

Reply:
Agreed. I think it'll be sufficient to check that at least one of the search results matches the given date.

Reply:
Is it still worth attempting to handle this edge case? It seems like with a high enough `--min-frame-coverage` value then it shouldn't be an issue, but maybe we don't want to rely on that?